editor/plugins/script_text_editor.cpp

```diff
@@ -1715,30 +1715,67 @@ void ScriptTextEditor::_text_edit_gui_input(const Ref<InputEvent> &ev) {
 			color_position.x = row;
 			color_position.y = col;
 
-			int begin = 0;
-			int end = 0;
-			bool valid = false;
+			int begin = -1;
+			int end = -1;
+			int expression_pattern = 0;
```

Comment:
Wouldn't this be better as an enum?

Reply:
Thanks for the review.
I have modified the code according to your suggestion.

Reply:
Looks good IMO, easier to read and understand now. I'm not a member of the Godot team so I can't officially review or accept PRs